docs-source/migrating-to-v7.md

```diff
@@ -0,0 +1,80 @@
+# Migration Guide
+
+## Migrating From v6 to v7
+
+There are a number of important changes in v7 including some that may be breaking depending on your scenarios. They are as follows:
+
+### ID Token Validation
+
+The authentication SDK now includes all-new ID Token validation capable of validating both RS256 and HS256 signed tokens.
+
+If your app is configured for:
+
+#### RS256
+
+No changes are required to your application.
+
+JWKS caching has changed. Previously it would cache the document once and it would be indefinitely until the app is restarted.
+
+The JWKS document will now only cache for 10 minutes and will retry more frequently when presented with new signing keys. This provides support for rotation of signing keys in the future.
+
+#### HS256
+
+- If using HS256 and your app is confidential (e.g. a web server) then you must set `SigningAlgorithm` to `SigningAlgorithm.HS256` on all `AuthenticationApiClient` requests.
+
+- If using HS256 and your app is NOT confidential (e.g. a desktop or mobile app) you should plan to move to RS256 as soon as possible. Client secrets can not be kept secure in these types of applications.
```

Comment:
What about adding a link to an article that talks the diffs between confidential / non confidential, like this one https://auth0.com/docs/applications/concepts/app-types-confidential-public